### PR TITLE
Deprecate the nonuniform attribute

### PR DESCRIPTION
**Pull request by proux01**
Now subsumed byt the #[warning="-uniform-inheritance"] attribute introduced in https://github.com/coq/coq/pull/16902 .
Following a remark from @Alizter in https://github.com/coq/coq/pull/17713

- [x] Added / updated **test-suite**.

- [x] Added **changelog**.
- [x] Added / updated **documentation**.
  
  - [ ] ~Documented any new / changed **user messages**.~
  - [ ] ~Updated **documented syntax** by running `make doc_gram_rsts`.~

- [x] Opened **overlay** pull requests.

### Overlays

- [ ] https://github.com/LPCIC/coq-elpi/pull/467
- [ ] https://github.com/Mtac2/Mtac2/pull/388

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

File: doc/sphinx/addendum/implicit-coercions.rst [hunk 1]
```diff
@@ -209,7 +213,6 @@ Coercion Classes
 
      The :ref:`test for ambiguous coercion paths <ambiguous-paths>`
      may yield false positives involving the coercion :token:`qualid`.
-     Use the :attr:`nonuniform` attribute to silence this warning.
```
Comment (Alizter) [inline doc/sphinx/addendum/implicit-coercions.rst:212]: Maybe better to give the warning example too.

**Comment by Alizter** [comment]
You could make `nonuniform` an "alias" of the warning attribute and also cleanup vernac now.

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by jfehrle** [comment]
I take it `uniform-inheritance` is not a new attribute but is only valid within `warnings`?  See #17717

**Comment by proux01** [comment]
> I take it `uniform-inheritance` is not a new attribute but is only valid within `warnings`? See #17717

Yes, it's the name of the warning the attribute `nonuniform` was silencing. That warning has been there for long now.

**Comment by proux01** [comment]
CI green

**Comment by proux01** [comment]
@Alizter would you be the assignee? this is ready

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**